tv_apps/smart-home/js/card_picker.js

```diff
@@ -180,18 +180,27 @@
         }
       });
 
-      cardList && cardList.forEach(card => {
+      cardList && cardList.forEach((card, index) => {
         var appButton = createButtonHelper(card);
         if (appButton) {
           appButton.dataset.parentType = 'empty';
-          candidates[this._getSortKey(card)] = appButton;
+          candidates.push({
+            index: index,
+            key: this._getSortKey(card),
+            button: appButton
+          });
         }
       });
 
-      var keys = Object.keys(candidates);
-      if (keys.length) {
-        keys.sort();
-        keys.forEach(key => appendToGridView(candidates[key]));
+      if (candidates.length) {
+        candidates.sort((a, b) => {
+          var compare = a.key.localeCompare(b.key);
+          if (!compare) {
+            return a.index - b.index;
```

Comment:
`a.key` and `b.key` is compared at line 197. This line is to make the sort stable.